lib/optimist.rb

```diff
@@ -246,6 +246,21 @@ def educate_on_error
     @educate_on_error = true
   end
 
+  ## Match long variables with inexact match.
+  ## If we hit a complete match, then use that, otherwise see how many long-options partially match.
+  ## If only one partially matches, then we can safely use that.
+  ## Otherwise, we raise an error that the partially given option was ambiguous.
+  def perform_inexact_match(arg, partial_match)  # :nodoc:
+    return @long[partial_match] if @long.has_key?(partial_match)
+    partially_matched_keys = @long.keys.grep(/^#{partial_match}/)
```

Comment:
Do we need to escape the partial match?  Command-line wise, I'm not sure if weird characters can get in there, but it might make sense to be defensive anyway.

```suggestion
    partially_matched_keys = @long.keys.grep(/^#{Regexp.escape(partial_match)}/)
```

Example of the difference:

```ruby
[1] pry(main)> partial_match = ".o"
[2] pry(main)> "foo".match?(/^#{partial_match}")
=> true
[3] pry(main)> "foo".match?(/^#{Regexp.escape(partial_match)}/)
=> false
```

Reply:
actually wondering if this should be a `select` with `start_with?`  regex may be overkill here